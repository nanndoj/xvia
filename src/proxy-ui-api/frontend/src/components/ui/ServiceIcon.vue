<template>
  <v-icon small :color="getServiceIconColor(service)">{{getServiceIcon(service)}}</v-icon>
</template>

<script lang="ts">
// Icon for a service. Shows lock icon with proper color.
import Vue from 'vue';

export default Vue.extend({
  props: {
    service: {
      type: Object,
      required: true,
    },
  },
  methods: {
    getServiceIcon(service: any): string {
      switch (service.ssl_auth) {
        case undefined:
          return 'mdi-lock-open-outline';
        case true:
          return 'mdi-lock';
        case false:
          return 'mdi-lock';
        default:
          return '';
      }
    },

    getServiceIconColor(service: any): string {
      switch (service.ssl_auth) {
        case undefined:
          return '';
        case true:
          return '#00e500';
        case false:
          return '#ffd200';
        default:
          return '';
      }
    },
  },
});
</script>
