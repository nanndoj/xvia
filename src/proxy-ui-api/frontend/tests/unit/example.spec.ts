import { mount, createLocalVue } from '@vue/test-utils';
import { SilenceWarnHack } from './silenceWarnHack';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
import ServiceIcon from '@/components/ui/ServiceIcon.vue';

const silenceWarnHack = new SilenceWarnHack();

const localVue = createLocalVue();
localVue.use(VueRouter);

describe('ServiceIcon', () => {
  let wrapper: any;
  let vuetify;

  const routes = [
    { path: '/clients', name: 'clients' },
  ];

  const router = new VueRouter({ routes });


  beforeEach(() => {

    silenceWarnHack.enable();
    vuetify = new Vuetify();
    localVue.use(Vuetify);
    silenceWarnHack.disable();

    wrapper = mount(ServiceIcon, {
      localVue,
      router,
      propsData: {
        service: {
          ssl_auth: true,
        },
      },
      mocks: {
        $t: () => 'localized text',
      },
    });
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  // `it' and `expect's ready to go now.
  it('snapshot test', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

});
