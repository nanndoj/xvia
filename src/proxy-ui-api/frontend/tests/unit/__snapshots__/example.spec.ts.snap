// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceIcon snapshot test 1`] = `
<i
  aria-hidden="true"
  class="v-icon notranslate mdi mdi-lock theme--light"
  style="font-size: 16px; color: rgb(0, 229, 0); caret-color: #00e500;"
/>
`;
